samples/hyperparameter_tunining.py

```diff
@@ -0,0 +1,216 @@
+                test_loss += F.cross_entropy(output, target, reduction='sum').item()
+                pred = output.argmax(dim=1, keepdim=True)
+                correct += pred.eq(target.view_as(pred)).sum().item()
+
+        logger.store(test_loss=test_loss / len(self.test_loader.dataset))
+        logger.store(accuracy=correct / len(self.test_loader.dataset))
+
+    def __call__(self):
+        for _ in self.loop:
+            self._train()
+            self._test()
+
+
+class LoaderConfigs(configs.Configs):
+    train_loader: torch.utils.data.DataLoader
+    test_loader: torch.utils.data.DataLoader
+
+
+class Configs(training_loop.TrainingLoopConfigs, LoaderConfigs):
+    epochs: int = 1
+
+    loop_step = 'loop_step'
+    loop_count = 'loop_count'
+
+    is_save_models = True
+    batch_size: int = 64
+    test_batch_size: int = 1000
+
+    use_cuda: bool = True
+    cuda_device: int = 0
+    seed: int = 5
+    train_log_interval: int = 10
+
+    is_log_parameters: bool = True
+
+    device: any
+
+    model: nn.Module
+
+    learning_rate: float = 0.01
+    momentum: float = 0.5
+    optimizer: optim.SGD
+
+    conv1_kernal: int
+    conv2_kernal: int
+
+    set_seed = 'set_seed'
+
+    main: MNIST
+
+
+@Configs.calc(Configs.device)
+def device(c: Configs):
+    from lab.util.pytorch import get_device
+
+    return get_device(c.use_cuda, c.cuda_device)
+
+
+def _data_loader(is_train, batch_size):
+    return torch.utils.data.DataLoader(
+        datasets.MNIST(str(logger.get_data_path()),
+                       train=is_train,
+                       download=True,
+                       transform=transforms.Compose([
+                           transforms.ToTensor(),
+                           transforms.Normalize((0.1307,), (0.3081,))
+                       ])),
+        batch_size=batch_size, shuffle=True)
+
+
+@Configs.calc([Configs.train_loader, Configs.test_loader])
+def data_loaders(c: Configs):
+    train = _data_loader(True, c.batch_size)
+    test = _data_loader(False, c.test_batch_size)
+
+    return train, test
+
+
+@Configs.calc(Configs.model)
+def model(c: Configs):
+    m: Net = Net(c.conv1_kernal, c.conv2_kernal)
+    m.to(c.device)
+    return m
+
+
+@Configs.calc(Configs.optimizer)
+def sgd_optimizer(c: Configs):
+    return optim.SGD(c.model.parameters(), lr=c.learning_rate, momentum=c.momentum)
+
+
+@Configs.calc(Configs.optimizer)
+def adam_optimizer(c: Configs):
+    return optim.Adam(c.model.parameters(), lr=c.learning_rate)
+
+
+@Configs.calc(Configs.set_seed)
+def set_seed(c: Configs):
+    torch.manual_seed(c.seed)
+
+
+@Configs.calc(Configs.loop_count)
+def loop_count(c: Configs):
+    return c.epochs * len(c.train_loader)
+
+
+@Configs.calc(Configs.loop_step)
+def loop_step(c: Configs):
+    return len(c.train_loader)
+
+
+def run(run_name: str, hparams: dict):
+    logger.set_global_step(0)
+
+    conf = Configs()
+    experiment = Experiment(name=run_name, writers={'sqlite', 'tensorboard'})
```

Comment:
I later realised that creating new experiment every-time adding multiple writers.